agent/app/agent_capability.go

```diff
@@ -37,6 +37,7 @@ const (
 	capabilityDockerPluginInfix                 = "docker-plugin."
 	attributeSeparator                          = "."
 	capabilityPrivateRegistryAuthASM            = "private-registry-authentication.secretsmanager"
+	capabilitySecretEnvSSM                      = "secrets-ssm-environment-variables"
```

Comment:
In that case, if there are environment variables and tmpfs support for secret provider `ssm` and only env var support for secret provider `xyz`, there will be 4 different capabilities as follows:
```
secrets-environment-variables
secrets-provider-ssm
secrets-tmpfs
secrets-provider-xyz
```

How will backend recognize if `xyz` supports tmpfs or not?

Reply:
> How will backend recognize if xyz supports tmpfs or not?

Does it matter? We have providers (ssm, xyz) and local mounts (env vars, tmpfs mounts). As far as I can tell, this should work for any combination.

My concern is that we add additional providers or mount points in the future and we have to extrapolate capabilities for *every combination* (providers * mounts) and not only the individual features (providers + mounts). 

Consider if we add support for S3 and Secrets Manager as providers and an http endpoint as a mount point.  That means that we need 9 capabilities instead of 6.

Reply:
I think if we leave the storage (environment variable and tmpfs) separate from the provider (ssm, xyz), then we're creating a requirement that we can use every combination of provider + storage.  If we ever break that, and say xyz can provide tmpfs but not environment variables, we then do not have a mechanism to indicate the capability in the future.

Reply:
> Does it matter? We have providers (ssm, xyz) and local mounts (env vars, tmpfs mounts). As far as I can tell, this should work for any combination.

Yes, it does. 

Say, agent 1.22.0 supports the following: 
- environment vairables and tmpfs for `ssm` 
- only environment variables for `xyz`

So the capabilities would be - 
```
secrets-environment-variables
secrets-tmpfs
secrets-provider-ssm
secrets-provider-xyz
```

Now say, agent 1.23.0 supports the following: 
- tmpfs support for `xyz`
The capabilities would still be - 
```
secrets-environment-variables
secrets-tmpfs
secrets-provider-ssm
secrets-provider-xyz
```

In this case, if I want to run a task with tmpfs support for `xyz`, how will backend know (according to capabilities) if agent 1.22.0 supports it or agent 1.23.0?

Reply:
> I think if we leave the storage (environment variable and tmpfs) separate from the provider (ssm, xyz), then we're creating a requirement that we can use every combination of provider + storage.

I was hoping this was our plan. It seems like we will be making arbitrary decisions about which providers support which mount types. Especially if we are only taking a text value and exposing it somewhere else.

Reply:
If we anticipate that this isn't the case, then you folks are right and this is the best way to do it.